### PR TITLE
Add `boostd index announce-latest` command

### PR DESCRIPTION
**Pull request by dirkmc**
```
$ boostd index announce-latest
Announced advertisement with cid baguqeeramxvzsv6txws3mtmahyijaf5trwbxjnkur3k44x7e4mgz57px4d6a

$ boostd index announce-latest-http
Announced advertisement to indexers over http with cid baguqeeras24lritlycfsvd3idryxamitoxh3bm6nbhvbj7xm3o6wv2dhfpmq
```

### CODE REVIEW

File: indexprovider/wrapper.go [hunk 1]
```diff
@@ -262,6 +263,14 @@ func (w *Wrapper) IndexerAnnounceAllDeals(ctx context.Context) error {
 	return merr
 }
 
+func (w *Wrapper) IndexerAnnounceLatest(ctx context.Context) (cid.Cid, error) {
+	e, ok := w.prov.(*engine.Engine)
```
Comment (masih) [inline indexprovider/wrapper.go:267]: Double checking `w.prov` will never be `nil`?

File: indexprovider/wrapper.go [hunk 2]
```diff
@@ -262,6 +263,14 @@ func (w *Wrapper) IndexerAnnounceAllDeals(ctx context.Context) error {
...
+	if !ok {
+		return cid.Undef, fmt.Errorf("index provider is disabled")
+	}
+	return e.PublishLatest(ctx)
```
Comment (masih) [inline indexprovider/wrapper.go:271]: There is a separate function for direct announce over HTTP. Should we check if direct announce is enabled and call that as well as this?

Comment (dirkmc) [reply indexprovider/wrapper.go:271]: Oh interesting.. I guess that could be a separate cli command so that we can debug both explicitly

Comment (masih) [reply indexprovider/wrapper.go:271]: or `--direct` flag?

File: indexprovider/wrapper.go (hunk 1 above)
Comment (dirkmc) [reply indexprovider/wrapper.go:267]: It will never be nil 👍

File: indexprovider/wrapper.go (hunk 2 above)
Comment (dirkmc) [reply indexprovider/wrapper.go:271]: I think I will add a separate command because you also have the option to explicitly specify the url

**Pull request merged**